### DISCUSSION TITLE
Customization of HTML email's styles

### DISCUSSION THREAD

**Discussion by Chircop97** [alfio-event/alf.io#1292] [General]
I noticed that support for html email have been already been configured -> [Pull Request](https://github.com/alfio-event/alf.io/pull/885)

However I noticed that there is no way to edit this from the administrator page, I managed to locate the files in mjml format in the alfio files, which editing this could get me to successfully edit the email's styling.

Can someone clarify if i'm mistaken or not or maybe some other way to add some styling to my emails?

**Comment by cbellone** [comment]
Your analysis is correct. 
At the moment, if you want to change the structure / styling of the HTML emails, you must modify the MJML template and build your own alf.io binary/docker image

**Reply by Chircop97** [reply]
Understood, will change that myself. Would be nice to have this as an improvement

**Reply by cbellone** [reply]
If you're willing to contribute, I'd be happy to discuss a possible implementation and review a PR! 👍

**Reply by Chircop97** [reply]
I was referring to modify the MJML cause I'm not familiar with Java. Would love to contribute

**Reply by Chircop97** [reply]
@cbellone So, I started trying to edit the mjml, however I noticed that the email I received is purely in plain text without any customizations at all. When I checked the email source, I noticed that the email body part was actually set to plain text among other types like calendar, but there were no content which was set to HTML. Could this be a bug were it is sent as plain text? How can I enforce it to use HTML emails?

**Reply by cbellone** [reply]
make sure that you don't have any override configuration for the template you're modifying.
You should check on the backoffice (configuration -> templates) at system, organization and event level

**Reply by Chircop97** [reply]
Understood, indeed that was it

**Reply by Chircop97** [reply]
So, I am trying to edit the mjml file. Any tips on the best way to implement this? Like is there a need for a restart for the changes to take affect? As of now some css was accepted but the editing to remove certain sections in the mjml didn't work.

**Reply by cbellone** [reply]
Yes, every modification requires a restart, because the MJML is converted to HTML (see `./build/generated/resources` folder) at build time.

We usually rely on the [MJML editor](https://mjml.io/try-it-live) for layout modifications and quick visual checking. This should reduce the feedback cycle
